base/abstractset.jl

```diff
@@ -476,6 +476,30 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Union{Real,Char}
+    (isempty(a) || isempty(b)) && return true
+    fa, la = first(a), last(a)
+    fb, lb = first(b), last(b)
+    if step(a) < 0
+        fa, la = la, fa
+    end
+    if step(b) < 0
+        fb, lb = lb, fb
+    end
+    not_overlapping = (la < fb) | (lb < fa)
+    if not_overlapping
+        return true
+    elseif abs(step(a)) === abs(step(b))
+        if fa in b || fb in a
+            return false
+        else
+            return true
```

Comment:
I'm not 100% sure this is valid (due to rounding). @timholy thoughts?

Reply:
Yeah, this is wrong: for example `isdisjoint(1e-100:.1:1, 0:.1:1)`

Reply:
Adding back `la in b || lb in a for a` will fix that, but not tricky examples like `isdisjoint(eps()/4:.1:.71, 0:.1:1)` which should return false since both contain the point `0.5` despite having the same step size and do not contain each-other's end points.

Reply:
If we can't get a great solution for Floats, perhaps split this into Integer an non-integer cases. For integers, this is valid (and when the step sizes are different, we can use the GCD to get an answer in O(1) no matter what). For non-integers, we can have a one-sided check
```julia
if fa in b || fb in a
    return false
else
    #slow path
```
or omit this check altogether.

Reply:
this makes sense.

Reply:
Would a fast branch for `(la - fa) < step(b) || (lb - fb) < step(a)` have the same rounding concerns for floats?

Reply:
isn't that just `isempty`? That should be valid. If either range is empty, they are disjoint.

Reply:
No, that checks if one range has a smaller range than the step of the other

Reply:
I'm not following. `isdisjoint(7:8, 1:6:12) == false` because `(last(7:8) - first(7:8)) < 6`. What is invalid?